source/extensions/tracers/xray/tracer.h

```diff
@@ -18,14 +21,18 @@ namespace Extensions {
 namespace Tracers {
 namespace XRay {
 
-static const char XRayTraceHeader[] = "x-amzn-trace-id";
+constexpr auto XRayTraceHeader = "x-amzn-trace-id";
```

Comment:
I am more confident the earlier form is optimal.

Reply:
How so? https://godbolt.org/z/rHUhf9

Reply:
I did not know about godbolt.org. That's awesome. My reasoning about `[]` being the more efficient form may not apply if you used `constexpr`. godbolt.org shows it's the exact same assembly emitted.

I was thinking that the `*` form provided an assignable pointer in addition to the static array, but evidently not so in your form. Thanks!